datafusion/expr/src/interval_arithmetic.rs

```diff
@@ -1469,6 +1472,8 @@ pub enum NullableInterval {
     MaybeNull { values: Interval },
     /// The value is definitely not null, and is within the specified range.
     NotNull { values: Interval },
+    /// Added to handle cases with insufficient statistics
+    Unknown,
 }
```

Comment:
`MaybeNull` with an unbounded interval corresponds to this Unknown variant, doesn't it ?

Reply:
That's an interesting idea. Do you mean we should use ScalarValue::Null if the value is absent?

Reply:
For me, Interval::MaybeNull{Null, Null} looks strange, honestly.
@jayzhan211 wdyt?

Reply:
Using ScalarValue::Null for representing unbounded intervals is discouraged. Although it's possible to construct these, performing arithmetic on them is not feasible. Could you detect the datatype and instead create ScalarValue::Datatype(None)? This approach is consistent with how we handle unbounded cases throughout the codebase.

Reply:
What I saw is that sometimes ColumnStatistics { ..., min, max, ... } both are equal to Precision::Absent. In such cases, we can't determine the type for Interval.

Reply:
I get the problem now. These unbounded intervals are meant to represent the uncertainty in the data. Adding another 'unknown' variant does seem like a burden, but you're right, it looks like the only way with the current ColumnStatistics. In the long run, probably the best fix is to add a DataType to the ColumnStatistic struct because I remember missing datatype info caused issues before too. But that's for another time, not this PR.

Reply:
We can file an issue to track the fix of it

Reply:
> What I saw is that sometimes ColumnStatistics { ..., min, max, ... } both are equal to Precision::Absent. In such cases, we can't determine the type for Interval.

Anywhere we have `ColumnStatistics` we should  also have the schema (and thus the `DataType` of all columns) -- maybe we just have to pass the schema along with the statistics?

Reply:
If I understand correctly, you're suggesting saving the entire schema for columns in the structs `Statistics` and `PartialJoinStatistics`, without passing it to `ColumnStatistics`. In my opinion, this is a good solution, particularly in cases where we seldom need the column type.

Reply:
When I tried to play with the stats, I attempted to get them from the `ExecutionPlan`. Here, there was a default absent value:

https://github.com/apache/datafusion/blob/c312ffe7d954563888a303beb8796848d20ff7c6/datafusion/core/src/physical_planner.rs#L470

Reply:
> I attempted to get them from the ExecutionPlan. Here, there was a default absent value:

I would expect that calling `execution_plan.statistics()` would match the schema returned from `execution_plan.schema()`

Reply:
Will re-check tomorrow then

Reply:
> Anywhere we have `ColumnStatistics` we should also have the schema (and thus the `DataType` of all columns) -- maybe we just have to pass the schema along with the statistics?

Yes, I think that's the right approach for this issue. Can we open a ticket for this? It should be an easy fix, and `ColumnStatistics` isn't used in many parts of the code.

Reply:
@alamb 
I have conducted some additional tests on various queries and observed the following results:

```
CREATE TABLE data_table (
    id INT,
    value INT
);
```

```
INSERT INTO data_table (id, value) VALUES
(1, 100),
(2, 200),
(3, 300),
(4, 400),
(5, 500),
(6, 600),
(7, 700),
(8, 800),
(9, 900),
(10, 1000);
```

```
SELECT id, value FROM data_table WHERE value > 500; 
```

_Log:_ 
`Schema: id: Int32, value: Int32` 
`Column 0: ColumnStatistics { null_count: Inexact(0), max_value: Exact(Int32(NULL)), min_value: Exact(Int32(NULL)), distinct_count: Absent }`
`Column 1: ColumnStatistics { null_count: Inexact(0), max_value: Inexact(Int32(NULL)), min_value: Inexact(Int32(501)), distinct_count: Absent }`

```
SELECT AVG(value) AS average_value, SUM(value) AS total_value FROM data_table; 
```

_Log:_
`Schema: average_value: Float64, total_value: Int64` 
`Column 0: ColumnStatistics { null_count: Absent, max_value: Absent, min_value: Absent, distinct_count: Absent }`
`Column 1: ColumnStatistics { null_count: Absent, max_value: Absent, min_value: Absent, distinct_count: Absent }`

```
SELECT a.id AS id_a, a.value AS value_a, b.id AS id_b, b.value AS value_b
FROM data_table a
CROSS JOIN data_table b;
```
_Log:_
`Schema: id_a: Int32, value_a: Int32, id_b: Int32, value_b: Int32`
`Column 0: ColumnStatistics { null_count: Exact(0), max_value: Absent, min_value: Absent, distinct_count: Absent }`
`Column 1: ColumnStatistics { null_count: Exact(0), max_value: Absent, min_value: Absent, distinct_count: Absent }`
`Column 2: ColumnStatistics { null_count: Exact(0), max_value: Absent, min_value: Absent, distinct_count: Absent }`
`Column 3: ColumnStatistics { null_count: Exact(0), max_value: Absent, min_value: Absent, distinct_count: Absent }`